src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
+	 * @param amount
+	 */
+	public void delayEventCheck(long amount)
+	{
+		long projectedDelayTime = Main.game.getMinutesPassed() + amount + GameCharacter.MINIMUM_POST_DELAY_DURATION;
+		if(projectedDelayTime > this.timeOfNextEventCheck)
+		{
+			this.timeOfNextEventCheck = projectedDelayTime;
+		}
+	}
+	
+	/**
+	 * Returns true if player has all rules at least at 0 obedience gain.
+	 */
+	public boolean isPlayerObeyingRules()
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getPerformanceQuality() < 0)
+			{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	/**
+	 * Handles player slavery updates
+	 */
+	public void handlePlayerSlavery(long minutesPassed)
+	{		
+		
+		boolean forcePunishment = false; // If true, forces the player to take punishment. Use for high obedience drops.
+		
+		// Checking rules compliance here
+		if(this.timeOfNextRuleCheck <= minutesPassed)
+		{
+			int totalRules = 0;
+			float obedienceChangePositive = 0;
+			float obedienceChangeNegative = 0;
+			float obedienceChangeTotal = 0;
+			
+			for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+				totalRules++;
+				float obedienceChange = slaveryRule.getPerformanceQuality();
+				if(obedienceChange > 0)
+				{
+					obedienceChangePositive += obedienceChange;
+				}
+				else
+				{
+					obedienceChangeNegative += obedienceChange;
+				}
+			}
+			
+			//If the player is disobeying a rule, they don't get positive gains from obedience.
+			if(obedienceChangeNegative == 0)
+			{
+				obedienceChangeTotal += obedienceChangePositive;
+			}
+			else
+			{
+				obedienceChangeTotal += obedienceChangeNegative;
+			}
+			
+			if(totalRules > 0)
+			{
+				obedienceChangeTotal /= totalRules; // The more rules are set for the player, the less they individually impact the result.
+			}
+			
+			Main.game.getPlayer().incrementObedience(obedienceChangeTotal, false);
+			this.timeOfNextRuleCheck += 60;
+			
+			// Checking to see if the player deserves a punishment for being a bad [player.gender] ;3
+			if(this.obedience < GameCharacter.FORCED_PUNISHMENT_THRESHOLD || obedienceChangeTotal < GameCharacter.FORCED_PUNISHMENT_DROP_TRIGGER)
+			{
+				forcePunishment = true;
+			}
+		}
+		
+		// Checking for event spawning here
+		if(this.timeOfNextEventCheck <= minutesPassed && Main.game.getPlayer().isWithinOwnersPropery() && Main.game.getActiveNPC() == null)
+		{
+			
+			// At 0 weight, an event won't be generated. If all events return 0 weight, then no event happens.
+			Main.game.setActiveNPC((NPC)this);
+			int currentCandidateWeight = 0;
+			PlayerSlaveryEvent currentCandidate = null;
+			
+			if(this.getSlaveryEvents() != null)
+			{
+				for(PlayerSlaveryEvent slaveryEvent : this.getSlaveryEvents()) {
+					if(!forcePunishment || (forcePunishment && slaveryEvent.getIsPunishment())) // Making sure to comply with punishment forcing.
```

Comment:
```
!A || (A && B)
A B !A RESULT
0 0  1  1
0 1  1  1
1 0  0  0
1 1  0  1
```

You should check for not A or B. So if (!forcePunishment || slaveryEvent.getIsPunishment())

Reply:
Done